src/main/java/it/pagopa/pm/gateway/service/CcResumeService.java

```diff
@@ -0,0 +1,196 @@
+import java.util.Map;
+import java.util.Objects;
+
+import static it.pagopa.pm.gateway.constant.Messages.GENERIC_ERROR_MSG;
+import static it.pagopa.pm.gateway.constant.Messages.PATCH_CLOSE_PAYMENT_ERROR;
+import static it.pagopa.pm.gateway.constant.VposConstant.RESULT_CODE_AUTHORIZED;
+import static it.pagopa.pm.gateway.constant.VposConstant.RESULT_CODE_CHALLENGE;
+import static it.pagopa.pm.gateway.dto.enums.PaymentRequestStatusEnum.*;
+import static it.pagopa.pm.gateway.dto.enums.TransactionStatusEnum.TX_AUTHORIZED_BY_PGS;
+import static it.pagopa.pm.gateway.dto.enums.TransactionStatusEnum.TX_REFUSED;
+
+@Service
+@Slf4j
+public class CcResumeService {
+
+    @Value("${vpos.requestUrl}")
+    private String vposUrl;
+
+    @Autowired
+    private PaymentRequestRepository paymentRequestRepository;
+
+    @Autowired
+    private RestapiCdClientImpl restapiCdClient;
+
+    @Autowired
+    private VPosRequestUtils vPosRequestUtils;
+
+    @Autowired
+    private VPosResponseUtils vPosResponseUtils;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    public void startResume(CreditCardResumeRequest request, String requestId) {
+        PaymentRequestEntity entity = paymentRequestRepository.findByGuid(requestId);
+
+        if (Objects.isNull(entity)) {
+            log.error("No CreditCard request entity has been found for requestId: " + requestId);
+            return;
+        }
+
+        if (Objects.nonNull(entity.getAuthorizationOutcome())) {
+            log.warn(String.format("requestId %s already processed", requestId));
+            entity.setErrorMessage("requestId already processed");
+            return;
+        }
+        processResume(request, entity, requestId);
+    }
+
+    private void processResume(CreditCardResumeRequest request, PaymentRequestEntity entity, String requestId) {
+        String methodCompleted = request.getMethodCompletd();
+        try {
+            StepZeroRequest stepZeroRequest = objectMapper.readValue(entity.getJsonRequest(), StepZeroRequest.class);
+            stepZeroRequest.setIsFirstPayment(false);
+            MethodCompletedEnum methodCompletedEnum = (StringUtils.isBlank(methodCompleted)) ? null : MethodCompletedEnum.valueOf(methodCompleted);
+            if (entity.getResponseType().equalsIgnoreCase(ThreeDS2ResponseTypeEnum.METHOD.name())) {
+                Map<String, String> params = createRequestForStep1(methodCompletedEnum, entity, stepZeroRequest);
+                executeStep1(params, entity, stepZeroRequest);
+            }
+        } catch (Exception e) {
+            log.error(String.format("error during execution of resume for requestId %s, stackTrace: %s ", requestId, Arrays.toString(e.getStackTrace())));
+        }
+    }
+
+    private Map<String, String> createRequestForStep1(MethodCompletedEnum methodCompletedEnum, PaymentRequestEntity entity, StepZeroRequest stepZeroRequest) throws IOException {
```

Comment:
Maybe this method is not useful since you are calling vPosRequestUtils.buildStepOneRequestParams() to actually create the request object